### PR TITLE
feat: eslint config and husky

### PR DESCRIPTION
**Pull request by braianj**

### CODE REVIEW

File: tsconfig.json [hunk 1]
```diff
@@ -18,7 +19,8 @@
     "strict": true,
     "noUnusedLocals": true,
     "noUnusedParameters": true,
-    "noFallthroughCasesInSwitch": true
+    "noFallthroughCasesInSwitch": true,
+    "allowSyntheticDefaultImports": true
```
Comment (LautaroPetaccio) [inline tsconfig.json:23]: Would you mind explaining why these changes were needed?

File: .husky/pre-commit [hunk 2]
```diff
@@ -0,0 +1,3 @@
+npm run fix:code
+npm run fix:prettier
+npm run test
```
Comment (LautaroPetaccio) [inline .husky/pre-commit:3]: Would you mind removing the tests here?

File: .husky/pre-commit [hunk 3]
```diff
@@ -0,0 +1,3 @@
+npm run fix:code
+npm run fix:prettier
```
Comment (LautaroPetaccio) [inline .husky/pre-commit:2]: Isn't prettier ran through eslint when running the plugins?

**Review by LautaroPetaccio — COMMENTED** [review]
Great job! I've left some comments to review, but it's great we have linting here

File: .husky/pre-commit (hunk 3 above)
Comment (braianj) [reply .husky/pre-commit:2]: I run those scripts to test it., and some times eslint make changes after ordering the import. Also CSS and JSON are not included into the `fix:code` command:
```
    "check:code": "eslint ./src --ext .ts,.tsx,.js,.jsx",
    "fix:code": "npm run check:code -- --fix",
```

File: tsconfig.json (hunk 1 above)
Comment (braianj) [reply tsconfig.json]: sure, because we use `import React from 'react'` instead of `import * as React from 'react'`, and that throw that error: `Unable to resolve path to module ‘react’   import/no-unresolved`
BUT this was solved with the setting in `eslintrc`


  ```
settings: {
    'import/resolver': {
      node: {
        extensions: ['.js', '.jsx', '.ts', '.tsx']
      }
    }
  } 
```

File: .husky/pre-commit [hunk 4]
```diff
@@ -0,0 +1,3 @@
+npm run fix:code
```
[the remark this answers is not in the document]
Comment (braianj) [reply .husky/pre-commit]: What do you think now?

[the remark this answers is not in the document]
Comment (LautaroPetaccio) [reply .husky/pre-commit]: Seems great now!

**Review by LautaroPetaccio — APPROVED** [review]
Great job 🚀
